users/app.py

```diff
@@ -60,5 +56,40 @@ def add_update_user():
     return jsonify(user_object), (201 if added_new_user else 200)
 
 
+def is_authorized_to_edit(username, users_collection):
+    """Queries the db to find authorization of the given user.
+
+    Documents in the users collection should look like
+        {"username": "cmei4444",
+        "name": "Carolyn Mei",
+        "is_organizer": True}
+    """
+    if users_collection.count_documents({"username": username}) == 0:
+        return False  # user not found
+    cursor = users_collection.find({"username": username})
+    for user in cursor:
```

Comment:
If you expect only a single document, you could also take the first item and then assert that the cursor was done.

This comes down to whether you expect to allow multiple documents with the same username, or if that indicates some sort of underlying logic error.

Reply:
Ideally, for authorization purposes we should have the identifying user information be unique to each user (I think OAuth 2.0 auto generates unique user IDs). The MongoDB collection.find query returns a cursor with all results and not just the first, so I think I'll switch to checking just the first result if that's what you mean.

Reply:
Done https://github.com/knative-portability/large-events/pull/2/commits/d5c146c0e92b1c86c7f14a7413fcf05905089807 https://github.com/knative-portability/large-events/pull/2/commits/b9ae63e97ced0c21d15a2164be1b3a534aec6760 https://github.com/knative-portability/large-events/pull/2/commits/ddcdbff0d00629f99071e41e21379025b350ae91